### PR TITLE
feat: ha groups assignations

### PR DESCRIPTION
**Pull request by achillemyette1**
This new role enables the operator to assign resources to existing HA groups.

### CODE REVIEW

File: roles/ha/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+You must provide the `proxmox_ha_resources` variable in order to assign
+ressources to existing HA Groups.
+
+Example:
+
+	---
+	proxmox_ha_resources:
+	  - group: "HA_all"
+		sid: "vm:110"
```
[the remark this answers is not in the document]
Comment (achillemyette1) [reply roles/ha/README.md]: fixed

File: roles/ha/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+- name: Assign ressources to HA Group
+  ansible.builtin.uri:
+    url: "https://{{ proxmox_api_host }}:8006/api2/json/cluster/ha/resources"
+    method: POST
+    body_format: json
+    status_code: [200]
+    return_content: true
+    validate_certs: no
```
Comment (btravouillon) [inline roles/ha/tasks/main.yml:9]: From a security standpoint, I believe certificates should be valid by default. Maybe a variable would help here? (default to yes)

File: roles/ha/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ressources to existing HA Groups.
+
+Example:
+
+    ---
+    proxmox_ha_resources:
+      - group: "HA_all"
+        sid: "vm:110"
+      - group: "HA_vn-c"
+        sid: "vm:108"
```
Comment (btravouillon) [inline roles/ha/README.md:15]: Just to be sure my understanding is correct: if one needs to configure 12 VMs in a single group, the structure would be the following?
```
proxmox_ha_resources:
  - group: "HA_group"
    sid: "vm:1"
  - group: "HA_group"
    sid: "vm:2"
  [...]
  - group: "HA_group"
    sid: "vm:12"
```

If so, could we consider a structure with group name and members?

```
proxmox_ha_resources:
  - group: "HA_group"
    members:
      - "vm-1"
      - "vm-2" 
      [...]
      - "vm-12"

**Review by btravouillon — COMMENTED** [review]
Nice feature. A couple of comments in-line.

File: roles/ha/README.md (hunk 3 above)
Comment (btravouillon) [reply roles/ha/README.md:15]: @pbouchez @ahmam any thoughts on what should be an appropriate structure here?

Comment (ahmam) [reply roles/ha/README.md:15]: based on the "This role DOES NOT creates the HA Groups! You must create them yourself." I see that the idea of having a group structure members will be the most appropriate in this context.

Comment (pbouchez) [reply roles/ha/README.md:15]: +1 for the group structure members

File: roles/ha/tasks/main.yml (hunk 2 above)
Comment (achillemyette1) [reply roles/ha/tasks/main.yml]: Done, i am not sur if there is a way to implement defaults at the collection level, so i created a defaults file at the role level.

File: roles/ha/README.md (hunk 3 above)
Comment (achillemyette1) [reply roles/ha/README.md]: Done !

**Review by btravouillon — APPROVED** [review]
LGTM 👍

**Review by pbouchez — APPROVED** [review]

**Review by pbouchez — APPROVED** [review]

**Pull request merged**